src/views/projectExplorer/index.ts

```diff
@@ -791,32 +791,45 @@ export default class ProjectExplorer implements TreeDataProvider<ProjectExplorer
       }),
       commands.registerCommand(`vscode-ibmi-projectexplorer.changeLibraryDescription`, async (element: Library) => {
         if (element) {
-          await commands.executeCommand(`code-for-ibmi.changeObjectDesc`, {
-            path: `${element.libraryInfo.library}/${element.libraryInfo.name}`,
-            type: element.libraryInfo.type.startsWith(`*`) ? element.libraryInfo.type.substring(1) : element.libraryInfo.type,
-            text: element.libraryInfo.text
-          });
-
+          await commands.executeCommand(`code-for-ibmi.changeObjectDesc`, toObjectBrowserObject(element));
           this.refresh();
         }
       }),
       commands.registerCommand(`vscode-ibmi-projectexplorer.copyLibrary`, async (element: Library) => {
         if (element) {
-          await commands.executeCommand(`code-for-ibmi.copyObject`, {
-            path: `${element.libraryInfo.library}/${element.libraryInfo.name}`,
-            type: element.libraryInfo.type.startsWith(`*`) ? element.libraryInfo.type.substring(1) : element.libraryInfo.type
+          const libraryName = await window.showInputBox({
+            prompt: l10n.t('Enter new library file name'),
+            placeHolder: l10n.t('New library name'),
+            validateInput: (name) => {
+              if (name.length > 10) {
+                return l10n.t('Library name must be 10 characters or less');
+              }
+              else if (name.toLocaleUpperCase() === element.libraryInfo.name.toLocaleUpperCase()) {
+                return l10n.t('New library name must be different from the existing one');
+              }
+              else {
+                return null;
+              }
+            }
           });
 
-          this.refresh();
+          if (libraryName) {
+            try {
+              const result = await ibmi!.getConnection().runCommand({ command: `CPYLIB FROMLIB(${element.libraryInfo.name}) TOLIB(${libraryName})` });
+              if (!result.code) {
+                if (await window.showInformationMessage(l10n.t('{0} successfully copied to {1}.', element.libraryInfo.name, libraryName), l10n.t("Add to library list"))) {
+                  commands.executeCommand("vscode-ibmi-projectexplorer.projectExplorer.addToLibraryList", {name: libraryName});
```

Comment:
Should there be an `await` here?

Reply:
Since we don't do anything after this, not necessarily. But I'll add it for consistency's sake.